jcommune-service/src/test/java/org/jtalks/jcommune/service/transactional/TransactionalGroupServiceTest.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (C) 2011  JTalks.org Team
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.jtalks.jcommune.service.transactional;
+
+import org.jtalks.common.model.entity.Group;
+import org.jtalks.common.service.exceptions.NotFoundException;
+import org.jtalks.common.validation.ValidationException;
+import org.jtalks.jcommune.model.dao.GroupDao;
+import org.jtalks.jcommune.model.dto.GroupAdministrationDto;
+import org.jtalks.jcommune.service.GroupService;
+import org.jtalks.jcommune.service.exceptions.OperationIsNotAllowed;
+import org.jtalks.jcommune.service.security.AdministrationGroup;
+import org.mockito.Mock;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.ExpectedExceptions;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static java.util.Collections.singletonList;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+import static org.testng.Assert.assertFalse;
+
+/**
+ * @author Pavel Vervenko
+ */
+public class TransactionalGroupServiceTest {
+
+    @Mock
+    private GroupDao groupDao;
```

Comment:
Maybe we could write a component test here? Not necessarily the one that uses MockMvc. Rather a test that initializes both DAO and HSQLDB and invokes service in a more real environment instead of mocking the DOCs.